### DISCUSSION TITLE
How to set the ｃａｂｌｅ between wind turbine and wind turbine

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1403] [Q&A]
Ｄｅａｒ　ｅｖｅｒｙｏｎｅ
How to set the link between the wind turbine and the wind turbine? For example, P1 wind turbine does not set the cable, and P8 fixed position sets the cable to P1 position. Is my understanding correct?
![image](https://user-images.githubusercontent.com/107483730/211557553-85079f65-b092-4ec6-b634-28bfa919d183.png)

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

I'm not familiar with this image, but are you referring to the modeling of a shared mooring system?  My interpretation of your image is that the floating offshore wind turbines (FOWTs) on the perimeter are anchored to the seafloor and the FOWTs on the edges also interconnect with FOWT P1 via shared moorings.

Are you trying to model this floating offshore wind farm in FAST.Farm?  With the release of FAST.Farm in OpenFAST v3.3, FAST.Farm can now model floating offshore wind farms with shared moorings.  Effectively, the mooring lines connecting FOWT P1 to the wind turbines on the edge of the perimeter are modeled in an instance of MoorDyn at the FAST.Farm level, and the mooring lines connecting the perimeter FOWTs to the seabed are modeled in instances of MoorDyn at the OpenFAST level.  See the following pull request--including links to updated FAST.Farm, MoorDyn, and API documentation--for more information: https://github.com/OpenFAST/openfast/pull/1086.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
Yes, your understand is right. I think the P1 cable is linked to the edge wind turbine, so I will describe what I should do now. The edge wind turbine cable does not need to be set, but only needs to set the anchor end of the P1 cable to the corresponding position of the edge wind turbine. For example The P8 wind turbine MooringDyn.dat only needs to set 6 cables, and the other 2 are not set, which is equivalent to leaving two positions for P1 cable connection, Right?

If so, what about 4X4? How do the middle 4 connect?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

I'll just clarify that FAST.Farm within OpenFAST v3.3 supports `NumTurbines+1` instances of MoorDyn, where `NumTurbines` is the number of wind turbines.  The `NumTurbines` instances of MoorDyn are associated with the `NumTurbines` instances of OpenFAST and are used to model the moorings that connect between each floater and the seabed.  The extra (`+1`) instance of MoorDyn is at the FAST.Farm level and is used to model moorings that connect between floaters, but don't anchor to the seabed.

In your image, it looks like all turbines except P1 have moorings connecting from the floater to the seabed; these moorings would be modeled within the instances of MoorDyn at the OpenFAST level.  The moorings connecting between the floaters would be modeled in the instance of MoorDyn at the FAST.Farm level.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
sample of FarmMoorDynFile the file containing shared mooring system input parameters #764
I saw SharedMoorFile on this forum, but now there is no such file "FarmMoorDyn. dat", so the openfastV3.3 you mentioned should not be implemented? So I want to realize the shared mooring function in V3.1. I don't know how to do it?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

The functionality to model shared mooring systems mentioned in #764 was completed and released in FAST.Farm within OpenFAST v3.3.  If you want use this functionality, I suggest upgrading from OpenFAST v3.1 to v3.3.

I know that a sample FAST.Farm model demonstrating the shared mooring functionality is under development and should be shared publicly soon.  Perhaps @mattEhall wants to comment on that.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @mattEhall 
I am implementing OpenfastV3.3 Farm and using the shared mooring function. Can you send relevant files to my mailbox? wjl473008299@qq.com. I am deeply grateful.

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
I still can't get a reply（FarmMoorDyn. dat） from mattEhall, so do you have any way to solve the problem of shared mooring system？

**Reply by jjonkman** [reply]
My understanding is that a sample FAST.Farm model demonstrating the shared mooring functionality is still under development and should be shared soon.  Perhaps @mattEhall wants to provide more information.
